src/components/FlawComments/FlawComments.vue

```diff
@@ -77,181 +79,91 @@ onMounted(async () => {
   emit('loadInternalComments');
 });
 
-async function handleCommentSave() {
+function startNewComment(type: CommentType) {
+  isAddingNewComment.value = true;
+  newCommentType.value = type;
+}
+
+async function handleCommentSave(newComment: string) {
   emit(
     'comment:addFlawComment',
-    newComment.value,
+    newComment,
     userStore.userName,
-    CommentType[selectedTab.value],
+    CommentType[newCommentType.value],
   );
-  newComment.value = '';
   isAddingNewComment.value = false;
 }
 
-const parseCommentDisplayText = (text: string) => parseJiraTags(linkify(sanitizeHtml(text)));
-
-function linkify(text: string) {
-  const bugzillaLink = `${osimRuntime.value.backends.bugzilla}/show_bug.cgi?id=`;
-
-  const bugzillaRegex = /\[bug (\d+)\]/g;
-  const jiraRegex = /\[([^|]+)\|([^\]]+)\]/g;
-
-  return text
-    .replace(bugzillaRegex, `<a target="_blank" href="${bugzillaLink}$1">[bug $1]</a>`)
-    .replace(jiraRegex, '<a target="_blank" href="$2">$2</a>');
+function getCommentType(comment: ZodFlawCommentType) {
+  if (props.publicComments.includes(comment)) {
+    return CommentType.Public;
+  } else if (props.privateComments.includes(comment)) {
+    return CommentType.Private;
+  } else if (props.internalComments.includes(comment)) {
+    return CommentType.Internal;
+  } else {
+    return CommentType.System;
+  }
 }
 
-function parseJiraTags(text: string) {
-  const jiraUserLink = `${osimRuntime.value.backends.jiraDisplay}/ViewProfile.jspa?name=`;
-  const jiraTagRegex = /\[~([^[\]]+)\]/g;
-
-  return text
-    .replace(jiraTagRegex, (match, p1) => `<a target="_blank" href="${jiraUserLink}${p1}">${p1}</a>`);
+function getCommentListWithTypes(list: ZodFlawCommentType[]) {
+  return list.map(c => ({ ...c, type: getCommentType(c) }));
```

Comment:
I think I would prefer to add the `type` directly in Zod and change the filter functions used in `useFlawComments` to refine function in Zod, we have all the data we need there.